dbt/include/fabric/macros/adapters/relation.sql

```diff
@@ -83,6 +80,74 @@
+      -- Getting constraints from the old table
+      {% call statement('get_table_constraints', fetch_result=True) %}
+        SELECT DISTINCT Contraint_statement FROM
+        (
+          SELECT DISTINCT
+          CASE
+              WHEN tc.CONSTRAINT_TYPE = 'PRIMARY KEY'
+                  THEN 'ALTER TABLE <<REPLACE TABLE>> ADD CONSTRAINT PK_<<CONSTRAINT NAME>>_'+ccu.COLUMN_NAME+' PRIMARY KEY NONCLUSTERED('+ccu.COLUMN_NAME+') NOT ENFORCED'
+              WHEN tc.CONSTRAINT_TYPE = 'UNIQUE'
+                  THEN 'ALTER TABLE <<REPLACE TABLE>> ADD CONSTRAINT UK_<<CONSTRAINT NAME>>_'+ccu.COLUMN_NAME+' UNIQUE NONCLUSTERED('+ccu.COLUMN_NAME+') NOT ENFORCED'
+              END AS Contraint_statement
+          FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS tc INNER JOIN
+              INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE ccu
+                  ON tc.CONSTRAINT_NAME = ccu.CONSTRAINT_NAME
+          WHERE tc.TABLE_NAME = '{{ from_relation.identifier }}' and tc.TABLE_SCHEMA = '{{ from_relation.schema }}'
+          UNION ALL
+          SELECT
+            'ALTER TABLE <<REPLACE TABLE>> ADD CONSTRAINT FK_<<CONSTRAINT NAME>>_'+CU.COLUMN_NAME+' FOREIGN KEY('+CU.COLUMN_NAME+') references '+PK.TABLE_SCHEMA+'.'+PK.TABLE_NAME+' ('+PT.COLUMN_NAME+') not enforced'   AS Contraint_statement
+            FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS C
+            INNER JOIN INFORMATION_SCHEMA.TABLE_CONSTRAINTS FK ON C.CONSTRAINT_NAME = FK.CONSTRAINT_NAME
+            INNER JOIN INFORMATION_SCHEMA.TABLE_CONSTRAINTS PK ON C.UNIQUE_CONSTRAINT_NAME=PK.CONSTRAINT_NAME
+            INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE CU ON C.CONSTRAINT_NAME = CU.CONSTRAINT_NAME
+            INNER JOIN (
+                SELECT i1.TABLE_NAME, i2.COLUMN_NAME, i1.TABLE_SCHEMA, i2.TABLE_SCHEMA AS CU_TableSchema
+                FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS i1
+                    INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE i2 ON i1.CONSTRAINT_NAME =i2.CONSTRAINT_NAME
+                WHERE i1.CONSTRAINT_TYPE = 'PRIMARY KEY'
+            ) PT ON PT.TABLE_NAME = PK.TABLE_NAME AND PT.TABLE_SCHEMA = PK.TABLE_SCHEMA AND PT.CU_TableSchema = PK.TABLE_SCHEMA
+          WHERE FK.TABLE_NAME = '{{ from_relation.identifier }}' and FK.TABLE_SCHEMA = '{{ from_relation.schema }}'
+          and PK.TABLE_SCHEMA = '{{ from_relation.schema }}' and PT.TABLE_SCHEMA = '{{ from_relation.schema }}'
+        ) T WHERE Contraint_statement IS NOT NULL
+      {% endcall %}
+
+      {%call statement('drop_table_constraints', fetch_result= True)%}
+        SELECT drop_constraint_statement FROM
+        (
+          SELECT 'ALTER TABLE ['+TABLE_SCHEMA+'].['+TABLE_NAME+'] DROP CONSTRAINT ' + CONSTRAINT_NAME AS drop_constraint_statement
+          FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS
+          WHERE TABLE_NAME = '{{ from_relation.identifier }}' and TABLE_SCHEMA = '{{ from_relation.schema }}'
+        ) T WHERE drop_constraint_statement IS NOT NULL
+
+      {% endcall %}
```

Comment:
whoa -- this looks like it took a lot of time to work out! is this all necessary for renaming a relation?

Reply:
SP_RENAME is not supported yet. I am using CTAS and DROP TABLE to manage SP_RENAME. However, the contraints are created on __dbt_tmp/__dbt_backup relations but on final models. Hence, I have to re-create constraints on the final table before dropping __dbt_tmp/__dbt_backup relations.

This is not necessary as soon as SP_RENAME is supported (expecting to be available by September).